scripts/update_network_account_listOfChanges.ts

```
@@ -0,0 +1,190 @@
+  // sharding: {
+  //   nodesPerConsensusGroup: number
+  //   executeInOneShard: boolean
+  // }
+}
+
+const defaultConfig: ConfigSchema = {
+  p2p: {
+    baselineNodes: 1280,
+    minNodes: 1280,
+    maxNodes: 1280,
+  },
+  // Add default values for new configurations
+  // sharding: {
+  //   nodesPerConsensusGroup: 1280,
+  //   executeInOneShard: false,
+  // },
+}
+
+// Readline interface for terminal input
+const rl = readline.createInterface({
+  input: process.stdin as NodeJS.ReadableStream,
+  output: process.stdout as NodeJS.WritableStream,
+})
+
+const askQuestion = (query: string): Promise<string> => {
+  return new Promise((resolve) => rl.question(query, resolve))
+}
+
+// Utility function to validate and parse input based on expected type
+const parseInput = (value: string, expectedType: string, defaultValue: any): any => {
+  if (value === '') return defaultValue
+
+  switch (expectedType) {
+    case 'number':
+      const parsedNumber = parseFloat(value)
+      return isNaN(parsedNumber) ? defaultValue : parsedNumber
+    case 'boolean':
+      return value.toLowerCase() === 'true'
+    case 'string':
+      return value
+    default:
+      return defaultValue
+  }
+}
+
+// Function to dynamically ask configuration questions
+const askConfigQuestions = async (): Promise<ConfigSchema> => {
+  const userConfig: Partial<ConfigSchema> = {}
+
+  for (const section in defaultConfig) {
+    userConfig[section] = {}
+    for (const key in defaultConfig[section]) {
+      let attempts = 0
+      const maxAttempts = 3
+      let isValid = false
+
+      while (!isValid && attempts < maxAttempts) {
+        const value = await askQuestion(`Enter ${section}.${key} (default: ${defaultConfig[section][key]}): `)
+        const expectedType = typeof defaultConfig[section][key]
+        const parsedValue = parseInput(value, expectedType, defaultConfig[section][key])
+
+        if (parsedValue !== defaultConfig[section][key] || value === '') {
```

`parsedValue == defaultConfig[section][key]` this can be a genuine case right? I mean not an error. We can suppress this error or simply say you have not updated the value.

Also what's the purpose of `value === ''` condition?

---

The value === '' condition is checking for when the user just hits enter without typing anything, indicating they want to accept the default value.